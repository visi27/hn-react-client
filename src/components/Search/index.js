import { connect } from 'react-redux';
import { resetSearch, setSearch, submitSearch } from '../../actions/search';
import Search from './presenter';

function mapStateToProps(state) {
  const { searchTerm } = state.search;
  return {
    searchTerm,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onSearchChange: event => dispatch(setSearch(event.target.value)),
    onSearchSubmit: () => dispatch(submitSearch()),
    onReset: () => dispatch(resetSearch()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);
