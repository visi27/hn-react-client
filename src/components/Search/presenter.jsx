import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import Button from '../Button/resetButton';

class Search extends Component {
  componentDidMount() {
    if (this.textInput) {
      this.textInput.focus();
    }
  }

  render() {
    const { children, searchTerm, onSearchChange } = this.props;

    return (
      <form onSubmit={e => e.preventDefault()}>
        {children}
        <input
          type="text"
          defaultValue={searchTerm}
          onChange={onSearchChange}
          ref={(input) => {
            this.textInput = input;
          }}
        />
        <Button>RESET</Button>
      </form>
    );
  }
}

Search.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  searchTerm: PropTypes.string,
  onSearchChange: PropTypes.func.isRequired,
};

Search.defaultProps = {
  searchTerm: '',
  children: 'Search',
};

export default Search;
