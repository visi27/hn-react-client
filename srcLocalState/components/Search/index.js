import React, { Component } from 'react';
import Button from '../Button/';
import PropTypes from 'prop-types';

class Search extends Component {
  componentDidMount() {
    if(this.textInput) {
      this.textInput.focus();
    }
  }

  render () {
    const {searchTerm, onSearchChange, onSearchSubmit, onReset, children} = this.props;

    return (
      <form onSubmit={onSearchSubmit}>
        {children}
        <input type="text" value={searchTerm} onChange={onSearchChange}
               ref={(input) => {this.textInput = input;}}/>
        <Button onClick={onReset}>RESET</Button>
      </form>
    );
  }
}

Search.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  onSearchSubmit: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  children: PropTypes.string,
};

export default Search;
