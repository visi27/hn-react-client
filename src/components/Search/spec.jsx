import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import Search from './presenter';

configure({ adapter: new Adapter() });

describe('Search', () => {
  const props = {
    searchTerm: 'redux',
    onSearchChange: () => {},
    onSearchSubmit: () => {},
    onReset: () => {},
    children: 'Search',
  };

  test('has a valid snapshot', () => {
    const component = shallow(<Search {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
