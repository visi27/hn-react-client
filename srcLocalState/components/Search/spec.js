import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './';

describe('Search', () => {

  const props = {
    searchTerm: 'redux',
    onSearchChange: () => {},
    onSearchSubmit: () => {},
    onReset: () => {},
    children: 'Search'
  };

  it('renders without cashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search {...props} />, div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Search {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
