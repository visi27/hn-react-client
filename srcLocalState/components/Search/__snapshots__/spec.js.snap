// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  Search
  <input
    onChange={[Function]}
    type="text"
    value="redux"
  />
  <button
    className="button-inline"
    onClick={[Function]}
    type="button"
  >
    RESET
  </button>
</form>
`;
